### PR TITLE
Create a GitHub action to run maven tests

### PR DESCRIPTION
**Pull request by jrha**

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Maven
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (wpoely86) [inline .github/workflows/maven.yml:7]: same remark, can you change this to:
```
on: [push, pull_request]
```

**Comment by wpoely86** [comment]
@jrha I think you're hitting the same issue as I did with the tests for the template. You must first download all of the maven dependencies and then run the tests. For some reason, downloads at a later stage hang or fail.
I fixed it by running:
```
mvn_run "dependency:resolve $MVN_ARGS" && mvn_test
```

**Comment by jrha** [comment]
I think you may be right, but I was trying to build panc in a clean environment, oddly it is falling over at:
> `Error:  Failed to execute goal on project panc-rpm: Could not resolve dependencies for project org.quattor.pan:panc-rpm:rpm:10.8-SNAPSHOT: Could not find artifact org.quattor.pan:panc:zip:10.8-SNAPSHOT in clojars.org (https://repo.clojars.org/) -> [Help 1]`

Which it shouldn't be looking for, because that is what it should be building.
Nor should it be looking in the clojars repo, which makes me think it tried everything else first.

**Comment by jrha** [comment]
Aha! It fails the same way on my desktop now.

**Comment by jrha** [comment]
Ok, it's a circular dependency in the build process, the secret is to run `install` as well as `test`.

**Review by wpoely86 — APPROVED** [review]

**Pull request merged**